### ISSUE TITLE
`jp-OutputArea-child` breaks matplotlib inline in jupyter notebooks

### ISSUE THREAD

**Issue by davidkell** [Kitware/trame-jupyter-extension#9]
The latest [commit](https://github.com/Kitware/trame-jupyter-extension/commit/c6aa04028f6c7991b5c0f0e2ae7f00c4e1014210) for 2.1.3 release, updating `jp-OutputArea-child` to have height: 100%, breaks matplotlib inline.

Example screenshot of issue:

<img width="727" alt="image" src="https://github.com/user-attachments/assets/e7568073-fc87-464d-9422-34640b082372" />

And fixing it by removing the CSS:

<img width="669" alt="image" src="https://github.com/user-attachments/assets/6d16e837-94f4-4808-924c-5f861d04c7f7" />

Instructions to reproduce:

```python
# setup python environment with latest stable version, e.g. using conda
conda create -n trame-debug python=3.13
conda activate trame-debug

# install latest version of packages
pip install jupyter==1.1.1 trame-jupyter-extension==2.1.3 matplotlib==3.10.0

jupyter notebook

# inside notebook
import matplotlib.pyplot as plt
plt.plot([1, 2, 3, 4])
```

Happy submit PR but I just don't know the original intention of the CSS update.

**Comment by jourdain** [comment]
Thanks for reporting the issue. 

The intention of the CSS change in the trame extension is to allow a cell output to be detached and grow in height rather than being stuck at 600px height. 

To test that part in trame, you can do the following:

```
pip install trame trame-vtk trame-vuetify
```

Inside a cell

```python
from trame.app.demo import Cone

app = Cone()
await app.ui.ready

app.ui
```

Then right click on the left of the output cell with the UI to make it its own tab. 
Normally if the CSS is good, the 3D cone app should properly stretch vertically for use the available space. 

If you find a good solution for both, a PR would be great.

Thanks,

Seb

**Comment by shenberg** [comment]
I was hit with the same bug when using trame-jupyter-extension for a remote Jupyter Lab. Too bad I only found this place after spending a few hours debugging the problem and finding the offending CSS independently. I refined the CSS selector as:
```css
.jp-LinkedOutputView .jp-OutputArea-child {
    height: 100%;
}
```
To only set height: 100% for jp-outputArea-child that is contained in a linked output view (when you right-click a cell and choose 'Create new view for cell output').

Having said that, I think the correct solution would be to also add a `:has` pseudo-selector to make sure that the referenced jp-OutputArea-child actually contains the trame iframe, so something along the lines of:

```css
.jp-LinkedOutputView .jp-OutputArea-child:has(iframe#trame_trame__template_main) {
    height: 100%;
}
```
I'm not a CSS pro, how I read this selector is "element with class `jp-OutputArea-child`, which contains an iframe with id `trame_trame__template_main`, and is contained in an element with class `jp-LinkedOutputView`".

I didn't open a pull request because I don't know that `jp-LinkedOutputView` containing the `jp-OutputArea-child` is a narrow-enough or wide-enough filter (e.g. I don't use Jupyter Notebook anymore, only Jupyter Lab), and I don't know whether counting on the id of the iframe makes sense - I just pulled the id of the trame output iframe from the browser devtools, I don't know if that's actually a reliable and consistent indicator, rather I'm guessing it isn't, but I hope it can point in the direction of an actual solution (e.g. a trame-specific class added to the iframe)
